assets/js/src/segments/subscribers_calculator.ts

```diff
@@ -0,0 +1,103 @@
+import MailPoet from 'mailpoet';
+import _ from 'lodash';
+
+export interface FormItem {
+  wordpressRole: string | undefined;
+  segmentType: string | undefined;
+  action: string | undefined;
+  newsletter_id: string | undefined;
+  category_id: string | undefined;
+  product_id: string | undefined;
+}
+
+export interface Result {
+  count: number;
+  errors: string[];
+}
+
+interface ApiVersionWindow extends Window {
+  mailpoet_api_version: string;
+}
+
+declare let window: ApiVersionWindow;
+
+class SubscribersCalculator {
```

Comment:
I used class because I wanted to store a previous state. I don't want to call API when the name of the description is changed. I would be glad when you help me with a better solution.

Reply:
you can do that even with a function:
```
let previous = undefined;

function SubscribersCalculator() {
  if (previous) return previous;
  ... 
}
```